dask/dataframe/shuffle.py

```diff
@@ -134,33 +134,14 @@ def __dask_distributed_pack__(
             "name_input": self.name_input,
             "meta_input": to_serialize(self.meta_input),
             "parts_out": list(self.parts_out),
+            "concat_func": {"__module__": _concat.__module__, "__name__": "_concat"},
+            "getitem_func": {"__module__": "operator", "__name__": "getitem"},
+            "shuffle_group_func": {
+                "__module__": shuffle_group.__module__,
+                "__name__": "shuffle_group",
+            },
```

Comment:
Should we have a name for this? Indirect import? Quoted import? It's starting to feel like a new flavor of task, perhaps worth formalizing in some way?

It is *almost* not necessary to have this  in `dask/dataframe/shuffle` at this point, no?

Reply:
> Should we have a name for this? Indirect import? Quoted import? It's starting to feel like a new flavor of task, perhaps worth formalizing in some way?

Right - Once we nail down what we actually want this to look like, I agree that we should clearly document and possibly formalize this approach in some way.

> It is almost not necessary to have this in dask/dataframe/shuffle at this point, no?

Right.  I am still open minded to the idea of moving everything into `layers.py`.  However, I personally feel that there are a variety of reasons not to do this.  In general, I feel that the path forward is likely to be much easier if we only require the unpacking and materialization logic to live in the lightweight (`layers.py`) module.  For shuffle, the only stickling point is the empty pd/cudf.DataFrame metadata that we need within `_construct_graph`.  When the `__dask_distributed_pack__` method is defined within the `dask.dataframe` module, it is easy to simply serialize this object.  However, if that definition was forced to live in `layers.py`, we wouldn't really be able to use the robust DataFrame serialization machinery.  I think we could certainly brainstorm ways to get around this, but I suspect that similar challenges will keep popping up.

Reply:
Ooh, good point. That actually opens a lot of doors :)

Reply:
so, to clarify, we can indeed move the shuffle layers entirely into `layers.py`.  Should we go ahead and do this?  That file may eventually become quite bloated, but it shouldn;t be muich work to cross that bridge later.

Reply:
I went ahead and moved "everything" over to `layers.py` (this includes `ShuffleLayer`, `SimpleShuffleLayer`, `BlockwiseCreateArray`, and `BroadcastJoinLayer`).  I think you are right about keeping the entire Layer definition in one place :)

Reply:
+1 thanks @rjzamora!